wgpu-types/src/lib.rs

```diff
@@ -846,30 +1135,47 @@ pub enum PresentMode {
 }
 
 bitflags::bitflags! {
+    /// Different ways that you can use a texture.
+    ///
+    /// The usages determine what kind of memory the texture is allocated from.
```

Comment:
I'm very willing to bet that this impacts more than just where the texture is allocated, but I don't know enough to be sure